RDBC-460 Migrated 12 tests from Java client to Python client.

poissoncorp opened the pull request:
https://issues.hibernatingrhinos.com/issue/RDBC-460
The first part of the tests (12 out of ~90).

poissoncorp committed:
RDBC-460 Migrated 12 tests from Java client to Python client.
Made every test passing.

In document_session.py:
- Assured that _multi_load won't return None filled dictionary by filtering keys collection passed into function
- Load call without key list will return None, won't raise ValueError
- Added advanced.has_changes method checking if any document change occurred during session

In raven_commands.py:
- Fixed POST data formatting issues - Wrapped data into "Ids" dictionary key

gregolsky commented on pyravendb/commands/raven_commands.py:
```diff
@@ -109,7 +109,7 @@ def create_request(self, server_node):
             if (sum(len(x) for x in key_or_keys)) > 1024:
                 self.method = "POST"
-                self.data = list(key_or_keys)
+                self.data = dict(Ids=list(key_or_keys))
```
Can we adjust spacing around `=`?

ayende replied:
There is the general recommended `Python` formatting - `PEP8`.
We should probably use something like https://pypi.org/project/autopep8/ to ensure consistent formatting.

gregolsky replied:
Yup, something to run with github actions on PR.

gregolsky commented on pyravendb/store/document_session.py:
```diff
@@ -161,7 +161,7 @@ def _convert_and_save_entity(self, key, document, object_type, nested_object_typ
         if len(keys) == 0:
             return []
-
+        keys = list(filter(lambda x: x is not None,keys))
```
Spacing after comma.

gregolsky commented on pyravendb/tests/jvm_migrated_tests/test_what_changed.py:
```diff
@@ -0,0 +1,31 @@
+
+            user1, user2 = session.load(["users/1", "users/2"], User)
+
+            self.assertFalse(session._has_change(user1))
```
In the original test  it was `session.Advanced.HasChanged()`. Please add such method and replace the occurrences in these tests.

gregolsky replied:
https://github.com/ravendb/ravendb/blob/49086f551a2ec7349c0922f6ec96c189f94fb7a6/src/Raven.Client/Documents/Session/IAdvancedDocumentSessionOperations.cs#L54

gregolsky commented on pyravendb/tests/jvm_migrated_tests/test_load.py:
```diff
@@ -0,0 +1,113 @@
+class UserWithId(User):
+    def __init__(self,name,age,identifier=None):
+        super(UserWithId, self).__init__(name,age)
+        self.Id=identifier
```
spacing around `=`

gregolsky commented:
I think we should create a `v5.2` branch and merge these there.

ml054 commented on pyravendb/tests/jvm_migrated_tests/test_basic_document.py:
```diff
@@ -0,0 +1,62 @@
+        self.assertTrue('@metadata' in doc2.keys())
+        self.assertEqual(len(doc1), 1)
+        self.assertEqual(len(doc2), 1)
+
+
+
+
+
+
+
+
+
+
+
+
+
```
?

ChristianSauer commented:
Hi,
PEP8 is the basis for a lot of tools, but more and more current python
tools use black (psf/black: The uncompromising Python code formatter
(github.com) <https://github.com/psf/black> . It's more like eslint and
minimizes git diffs. If you have no tooling yet, I would recommend it over
other python linters.

Am Do., 8. Juli 2021 um 08:05 Uhr schrieb Ayende Rahien <
***@***.***>:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In pyravendb/commands/raven_commands.py
> <https://github.com/ravendb/ravendb-python-client/pull/89#discussion_r665893735>
> :
>
> > @@ -109,7 +109,7 @@ def create_request(self, server_node):
>              # If it is too big, we drop to POST (note that means that we can't use the HTTP cache any longer)
>              if (sum(len(x) for x in key_or_keys)) > 1024:
>                  self.method = "POST"
> -                self.data = list(key_or_keys)
> +                self.data = dict(Ids=list(key_or_keys))
>
> There is the general recommended Python formatting - PEP8.
> We should probably use something like https://pypi.org/project/autopep8/
> to ensure consistent formatting.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ravendb/ravendb-python-client/pull/89#discussion_r665893735>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABXKCSJ2JOM3NQPABCQZVQDTWU52LANCNFSM476TQ2AA>
> .
>


-- 

Christian Sauer |IT Consultant & Developer

codecentric AG | Hochstraße 11 | 42697 Solingen| Deutschland
tel: +49 (0) 212.233 628 30 | fax: +49 (0) 212.233 628 79 | mobil: +49 (0)
160
238 4840
www.codecentric.de | blog.codecentric.de | www.meettheexperts.de
<http://www.more4fi.de/>


Sitz der Gesellschaft: Solingen | HRB 25917| Amtsgericht Wuppertal
Vorstand: Rainer Vehns (Vorsitzender) . Ulrich Kühn . Lars Rückemann
Aufsichtsrat: Patric Fedlmeier (Vorsitzender) . Klaus Jäger . Jürgen Schütz

Diese E-Mail einschließlich evtl. beigefügter Dateien enthält vertrauliche
und/oder rechtlich geschützte Informationen. Wenn Sie nicht der richtige
Adressat sind oder diese E-Mail irrtümlich erhalten haben, informieren Sie
bitte sofort den Absender und löschen Sie diese E-Mail und evtl.
beigefügter Dateien umgehend. Das unerlaubte Kopieren, Nutzen oder Öffnen
evtl. beigefügter Dateien sowie die unbefugte Weitergabe dieser E-Mail ist
nicht gestattet.

gregolsky commented:
Thanks @ChristianSauer , we'll check it out.

poissoncorp committed:
RDBC-461 Added based on nodejs client .yml github action file running tests and formatting project with autopep8

poissoncorp committed:
RDBC-461 Created github actions .yml file running unit tests, fixed failing tests

test_advanced:
- Changed path format to output.txt to fit any environment

test_counters_operations & test_system_create_topology_files:
- Erased non-necessary print functions

test_server_operations:
- Added test databases cleanups

gregolsky commented on .github/workflows/RavenClient.yml:
```diff
@@ -0,0 +1,56 @@
+  push:
+    branches: [ RDBC-461 ]
+  pull_request:
+    branches: [ v5.0 ]
```
Let's do `v5.2` here
